src/App.tsx

```diff
@@ -9,39 +9,31 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as usersActions from './features/users';
+import { actions as selectedPostActions } from './features/selectedPost';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { hasError } = useAppSelector(state => state.users);
+  const { post: selectedPost } = useAppSelector(state => state.selectedPost);
+  const {
+    author,
+    loaded: postLoaded,
+    posts,
+    hasError: postError,
+  } = useAppSelector(state => state.author);
```

Comment:
Ensure that the `author` state slice exists in your Redux store. If it doesn't, accessing `state.author` will lead to runtime errors. Verify the structure of your Redux store to ensure this slice is correctly defined.